src/tree/remoteProject/SystemIdentityTreeItemBase.ts

```diff
@@ -0,0 +1,89 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { type Identity } from '@azure/arm-resources';
+import { createManagedServiceIdentityClient, createRoleDefinitionsItems, RoleDefinitionsTreeItem, type RoleDefinitionsItem } from '@microsoft/vscode-azext-azureutils';
+import { AzExtParentTreeItem, AzExtTreeItem, type IActionContext } from '@microsoft/vscode-azext-utils';
+import { localize } from '../../localize';
+import { type ManagedIdentityTreeItem } from './ManagedIdentityTreeItem';
+
+export type SystemIdentityTreeItemBase = SystemIdentityTreeItem | DisabledIdentityTreeItem;
+export function createSystemIdentityTreeItem(parent: ManagedIdentityTreeItem): SystemIdentityTreeItem | DisabledIdentityTreeItem {
+    if (!parent.parent.site.rawSite.identity?.type?.includes('SystemAssigned')) {
+        return new DisabledIdentityTreeItem(parent);
+    } else {
+        return new SystemIdentityTreeItem(parent, parent.parent.site.rawSite.identity);
+    }
+}
+class SystemIdentityTreeItem extends AzExtParentTreeItem {
+    public readonly identity: Identity;
+    public readonly parent: ManagedIdentityTreeItem;
+    public static contextValue: string = 'systemIdentity';
+    public state: string;
+
+    public constructor(parent: ManagedIdentityTreeItem, identity: Identity) {
+        super(parent);
+        this.identity = identity;
+        this.state = 'enabled';
+    }
+
+    public get description(): string | undefined {
+        return localize('enabled', 'Enabled');
+    }
+
+    public async loadMoreChildrenImpl(_clearCache: boolean, context: IActionContext): Promise<AzExtTreeItem[]> {
+        const msiClient = await createManagedServiceIdentityClient([context, this.subscription]);
+        const systemIdentity = await msiClient.systemAssignedIdentities.getByScope(this.parent.parent.id);
+        const roleDefinitionsItem: RoleDefinitionsItem[] = await createRoleDefinitionsItems(context, this.parent.subscription, systemIdentity)
+        return roleDefinitionsItem.map(rd => new RoleDefinitionsTreeItem(this, rd));
```

Comment:
Kind of a nit, but when User-Assigned Identity Tree Items and System-Assigned Identity Tree Items are added / enabled but have no role assignments, the tree view shows an expandable node which has nothing underneath it when expanded:

![image](https://github.com/user-attachments/assets/e712f2f1-7804-4ff3-8143-bd92084c35f9)

Users could be confused as to what this means.  Perhaps if an empty list is returned, we could default to a generic tree item stating no role assignments assigned yet?  Idk, thoughts?

Reply:
I'm not against the idea, but would definitely prefer this to be a part 2 iteration on the feature. It's definitely more of a nice to have feature.

Also for context, you can open the managed identity nodes in the portal where you can add roles. I may be able to open them directly to the add roles page (and do the same for system identity as well).